modules/gui/src/com/haulmont/cuba/gui/components/EditorWindowDelegate.java

```diff
@@ -151,7 +151,7 @@ public void setItem(Entity item) {
             }
             item = EntityCopyUtils.copyCompositions(item);
             handlePreviouslyDeletedCompositionItems(item, parentDs);
-        } else if (!PersistenceHelper.isNew(item)) {
+        } else if (!PersistenceHelper.isNew(item) && !entityStates.isLoadedWithView(item, ds.getView())) {
```

Comment:
Here we will have side effect. Reloading means that the new instance will not have property change listeners, that is desired behavior. We should check that it does not break existing projects.

Probably, it cannot be merged until 6.10.